src/extension_commands.ml

```diff
@@ -534,6 +534,116 @@ module Copy_type_under_cursor = struct
     command Extension_consts.Commands.copy_type_under_cursor handler
 end
 
+module Search_by_type = struct
+  let extension_name = "Search By Type"
+
+  let ocaml_lsp_doesnt_support_search_by_type instance ocaml_lsp =
+    match
+      Ocaml_lsp.is_version_up_to_date
+        ocaml_lsp
+        (Extension_instance.ocaml_version_exn instance)
+    with
+    | Ok () -> ()
+    | Error (`Msg msg) ->
+      show_message
+        `Warn
+        "The installed version of `ocamllsp` does not support type search. %s"
+        msg
+
+  let _get_search_results ~query ~limit ~with_doc text_editor client =
+    let doc = TextEditor.document text_editor in
+    let uri = TextDocument.uri doc in
+    let selection = TextEditor.selection text_editor in
+    Custom_requests.(
+      send_request
+        client
+        Type_search.request
+        (Type_search.make
+           ~uri
+           ~position:(Range.start (Selection.to_range selection))
+           ~limit
+           ~query
+           ~with_doc
+           ()))
+
+  let _search_by_type =
+    let handler (instance : Extension_instance.t) ~args:_ =
+      let search_by_type () =
+        match Window.activeTextEditor () with
+        | None ->
+          Extension_consts.Command_errors.text_editor_must_be_active
+            extension_name
+            ~expl:"The command to search for a value by it's type"
+          |> show_message `Error "%s" |> Promise.return
+        | Some _text_editor -> (
+          match Extension_instance.lsp_client instance with
+          | None ->
+            show_message `Warn "ocamllsp is not running" |> Promise.return
+          | Some (_client, ocaml_lsp)
+            when not (Ocaml_lsp.can_handle_search_by_type ocaml_lsp) ->
```

Comment:
@voodoos This boolean check results to `true` even when I have the opam switch's version of `ocaml-lsp-server` pinned to the branch that has the `TypeSearch Custom Requests` implementation.

Reply:
Does the LSP server advertise the necessary `handleTypeSearch` capability ?

Reply:
Does the LSP server advertise the necessary handleTypeSearch capability ?

Reply:
No it doesn't. 

```json
Result: {
    "capabilities": {
        "codeActionProvider": {
            "codeActionKinds": [
                "quickfix",
                "refactor.inline",
                "construct",
                "destruct (enumerate cases)",
                "inferred_intf",
                "put module name in identifiers",
                "remove module name from identifiers",
                "remove type annotation",
                "type-annotate"
            ]
        },
        "codeLensProvider": {
            "resolveProvider": false
        },
        "completionProvider": {
            "resolveProvider": true,
            "triggerCharacters": [
                ".",
                "#"
            ]
        },
        "declarationProvider": true,
        "definitionProvider": true,
        "documentFormattingProvider": true,
        "documentHighlightProvider": true,
        "documentSymbolProvider": true,
        "executeCommandProvider": {
            "commands": [
                "ocamllsp/view-metrics",
                "ocamllsp/open-related-source",
                "ocamllsp/merlin-jump-to-target",
                "ocamllsp/show-document-text",
                "ocamllsp/show-merlin-config",
                "dune/promote"
            ]
        },
        "experimental": {
            "ocamllsp": {
                "interfaceSpecificLangId": true,
                "handleSwitchImplIntf": true,
                "handleInferIntf": true,
                "handleTypedHoles": true,
                "handleWrappingAstNode": true,
                "diagnostic_promotions": true,
                "handleHoverExtended": true,
                "handleMerlinCallCompatible": true,
                "handleTypeEnclosing": true,
                "handleGetDocumentation": true,
                "handleConstruct": true
            }
        },
```

Reply:
Yes, I just found the bug :), more of an omission

Reply:
```json
"experimental": {
            "ocamllsp": {
                "interfaceSpecificLangId": true,
                "handleSwitchImplIntf": true,
                "handleInferIntf": true,
                "handleTypedHoles": true,
                "handleWrappingAstNode": true,
                "diagnostic_promotions": true,
                "handleHoverExtended": true,
                "handleMerlinCallCompatible": true,
                "handleTypeEnclosing": true,
                "handleGetDocumentation": true,
                "handleConstruct": true,
                "handleTypeSearch": true
            }
        },
```